### PR TITLE
#19746 : Major refactoring an improvement of very old code, removed a…

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
… lot of duped code, and started using the official list of "evil" SQL keywords for detecting dangerous code. The root cause of the problem was because the following code:

String conInode = fieldResourceName.substring(fieldResourceName.indexOf("/") + 1, fieldResourceName.indexOf("_"));

Specifically, the value of the fieldResourceName variable, which contains a format that is no longer applicable to the old code. This was causing the ViewTool to not work at all.

### CODE REVIEW

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/SQLResultsViewTool.java [hunk 1]
```diff
@@ -1,277 +1,278 @@
...
+
+    /**
+     * Scans the SQL query passed down by the user/developer looking for evil SQL words, as only {@code SELECT} queries
+     * (data reading operations) are allowed.
+     *
+     * @param sqlQuery The SQL query.
+     *
+     * @return If the SQL query is safe, returns {@code true}. Otherwise, returns {@code false}.
+     */
+    protected boolean containsEvilSqlWords(final String sqlQuery) {
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/SQLResultsViewTool.java:272]: I would move this method to the SQLUtil class and implement a test

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/SQLResultsViewTool.java [hunk 2]
```diff
@@ -1,277 +1,278 @@
...
+    /**
+     * Analyzes the words contained in the SQL query specified by the user/developer, and reports back when an invalid
+     * keyword or forbidden table is referenced.
+     *
+     * @param sql        The SQL query.
+     * @param contentlet The Contentlet containing the SQL code.
+     *
+     * @return If the SQL query contains no dangerous code, returns {@code true}. Otherwise, returns {@code false}.
+     */
+    protected boolean isSQLValid(final String sql, final Contentlet contentlet) {
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/SQLResultsViewTool.java:215]: I would add a test to this method

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/SQLResultsViewTool.java [hunk 3]
```diff
@@ -1,277 +1,278 @@
...
+     * @param dataSource    The database that the SQL query will run against.
+     * @param sql           The SQL query.
+     * @param parameterList The list of parameters that will be included in the query.
+     * @param startRow      The initial paginated record included in the result set.
+     * @param maxRow        The final paginated record included in the result set.
+     *
+     * @return The results returned by the SQL query.
+     */
+    public ArrayList<HashMap<String, String>> getParameterizedSQLResults(final String dataSource, final String sql,
+                                                                         final ArrayList<Object> parameterList, int
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/SQLResultsViewTool.java:98]: set to final

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/SQLResultsViewTool.java [hunk 4]
```diff
@@ -1,277 +1,278 @@
...
+    ArrayList<Object> parameterList, int startRow, int maxRow) {
+        if (dataSource.equals(DEFAULT_DATASOURCE) && !Config.getBooleanProperty
+                ("ALLOW_VELOCITY_SQL_ACCESS_TO_DOTCMS_DB", false)) {
+            return reportError(CollectionsUtils.map(DOT_CONNECT_SQL_ERROR, "SQLResultsViewTool failed to execute " +
+                    "query on default connection pool because ALLOW_VELOCITY_SQL_ACCESS_TO_DOTCMS_DB is set to false" +
+                    "."), "ALLOW_VELOCITY_SQL_ACCESS_TO_DOTCMS_DB is set to false.");
+        }
+        if (!UtilMethods.isSet(sql)) {
+            // SQL Query is not Set. So, return an empty list
+            return new ArrayList<>();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/SQLResultsViewTool.java:124]: Probably Collections.emptyList() would be better

**Review by jdotcms — APPROVED** [review]

**Pull request merged**